streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -519,6 +563,33 @@ private void handleTasksPendingInitialization() {
         }
     }
 
+    private void handleInitialTaskReuse(final Map<TaskId, Set<TopicPartition>> activeTasksToCreate,
+                                        final Map<TaskId, Set<TopicPartition>> standbyTasksToCreate,
+                                        final Map<TaskId, RuntimeException> failedTasks) {
+        final Map<Task, Set<TopicPartition>> initialStandbyTasksToRecycle = assignInitialTasks(activeTasksToCreate, logPrefix, topologyMetadata, changelogReader);
+        final Map<Task, Set<TopicPartition>> initialStandbyTasksToUse = assignInitialTasks(standbyTasksToCreate, logPrefix, topologyMetadata, changelogReader);
+
+        // if this was the last local thread to receive its assignment, close all the remaining Tasks, as they are not needed
+        stateDirectory.closeInitialTasksIfLastAssginedThread();
+
+        // recycle the initial standbys to active, and remove them from the set of actives that need to be created
+        if (!initialStandbyTasksToRecycle.isEmpty()) {
+            final Set<Task> tasksToCloseDirty = new HashSet<>();
```

Comment:
What is the purpose of this one? We pass it into `recycleTaskFromStateUpdater` but we never read it?

Reply:
It looks like I actually had a bug here, which I believe I've fixed in https://github.com/apache/kafka/pull/16922/commits/d326ea8142629200ab15c7f38a45e75e95738edd